src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -131,7 +131,7 @@ public function load(array $configs, ContainerBuilder $container)
                     'macvim' => 'mvim://open?url=file://%%f&line=%%l',
                     'emacs' => 'emacs://open?url=file://%%f&line=%%l',
                     'sublime' => 'subl://open?url=file://%%f&line=%%l',
-                    'phpstorm' => 'phpstorm://open?url=file://%%f&line=%%l',
+                    'phpstorm' => 'Darwin' === PHP_OS ? 'phpstorm://open?file=%%f&line=%%l' : 'phpstorm://open?url=file://%%f&line=%%l',
```

Comment:
checking the OS will indeed not play well with VMs, as phpstorm will run on the host, not on the VM.
We should have a separate config key IMO